### ISSUE TITLE
macaron should honor gogs log-levels

### ISSUE THREAD

**Issue by brechsteiner** [gogits/gogs#2426]
macaron has Trace as default loglevel, it should use the loglevel from app.ini config file

**Comment by Unknwon** [comment]
Thanks your feedback!

> macaron has Trace as default loglevel

I don't think Macaron has log level at all.

**Comment by Unknwon** [comment]
Thoughts?

**Comment by Unknwon** [comment]
Close due to lack of feedback.

**Comment by k0lter** [comment]
Here is the the feedback :

With this configuration :

```ini
[log]
ROOT_PATH = /var/log/gogs
MODE = file
BUFFER_LEN = 10000
LEVEL = Info
```

gogs (macaron) is still logging in syslog like the following :

```ini
Feb  5 21:56:12 web2 gogs[11494]: 2016/02/05 21:56:12 #033[1;36m[T] Doing: MirrorUpdate#033[0m
Feb  5 22:56:12 web2 gogs[11494]: 2016/02/05 22:56:12 #033[1;36m[T] Doing: MirrorUpdate#033[0m
Feb  5 14:40:17 web2 gogs[11494]: [Macaron] Started GET / for 192.168.2.6
Feb  5 14:40:17 web2 gogs[11494]: 2016/02/05 14:40:17 #033[1;34m[D] Session ID: faf378431f0c331b#033[0m
Feb  5 14:40:17 web2 gogs[11494]: 2016/02/05 14:40:17 #033[1;34m[D] CSRF Token: YmmgeTQlq9rIdNQT-r9yaxPpeZI6MTQ1NDY2NDMxNzk0MDg3MzY3OA==#033[0m
Feb  5 14:40:17 web2 gogs[11494]: 2016/02/05 14:40:17 #033[1;34m[D] Template: user/dashboard/dashboard#033[0m
Feb  5 14:40:17 web2 gogs[11494]: [Macaron] Completed / 200 OK in 15.483065ms
Feb  5 14:40:17 web2 gogs[11494]: [Macaron] Started GET /avatars/4 for 192.168.2.6
Feb  5 14:40:17 web2 gogs[11494]: [Macaron] [Static] Serving /4
Feb  5 14:40:17 web2 gogs[11494]: [Macaron] Completed /avatars/4 304 Not Modified in 65.712Âµs
Feb  5 14:40:28 web2 gogs[11494]: [Macaron] Started GET /blq/101 for 192.168.2.6
Feb  5 14:40:28 web2 gogs[11494]: 2016/02/05 14:40:28 #033[1;34m[D] Session ID: faf378431f0c331b#033[0m
Feb  5 14:40:28 web2 gogs[11494]: 2016/02/05 14:40:28 #033[1;34m[D] CSRF Token: YmmgeTQlq9rIdNQT-
Feb  5 14:40:47 web2 gogs[11494]: 2016/02/05 14:40:47 #033[1;34m[D] Detected encoding: utf-8 (fast)#033[0m
Feb  5 14:40:47 web2 gogs[11494]: 2016/02/05 14:40:47 #033[1;34m[D] Detected encoding: utf-8 (fast)#033[0m
Feb  5 14:41:35 web2 gogs[11494]: [Macaron] Started GET /js/jquery-1.11.3.min.js for 192.168.2.6
Feb  5 14:41:35 web2 gogs[11494]: [Macaron] [Static] Serving /js/jquery-1.11.3.min.js
Feb  5 14:41:35 web2 gogs[11494]: [Macaron] Completed /js/jquery-1.11.3.min.js 304 Not Modified in 60.464Âµs
Feb  5 14:41:35 web2 gogs[11494]: [Macaron] Started GET /css/font-awesome-4.4.0.min.css for 192.168.2.6
Feb  5 14:41:35 web2 gogs[11494]: [Macaron] [Static] Serving /css/font-awesome-4.4.0.min.css
Feb  5 14:41:35 web2 gogs[11494]: [Macaron] Completed /css/font-awesome-4.4.0.min.css 304 Not Modified in 35.894Âµs
Feb  5 14:41:35 web2 gogs[11494]: [Macaron] Started GET /css/semantic-2.1.6.min.css for 192.168.2.6
```

So there is 2 issues:
- Events with apparently Trace log level are logged with level Info
- Even if a log file is defined for logging, syslog is used

**Comment by Unknwon** [comment]
Thanks, but

> I don't think Macaron has log level at all.

**Comment by technodrome** [comment]
This is probably because your Gogs installation is starting through a systemd .service file. As systemd is quite new to me, I cannot tell you how to reroute Macaron logs, but I do not agree with @Unknwon that there are no logs. For the moment, the only way I found to prevent Macaron from writing to syslog is to edit the gogs.service file in /lib/systemd/system. The problem, in my point of view is that the logs are not going to gogs/log/httpd.log (or serv.log), but to syslog, which is incredibly messy. I am a fan of separate logfiles.

**To silence Macaron logs via systemd:**
Edit file /lib/systemd/system/gogs.service and in the section `[Service]` add
`StandardOutput=null`

Then, as the systemd .service file was changed on the disk, reload the systemd daemon:
`systemctl daemon-reload`
